src/pages/Home.jsx

```diff
@@ -19,73 +59,79 @@ function Home() {
+                  className={classNames("btn btn-sm", {"btn-outline-primary": !article.favorited })} 
+                  
                   disabled={false}
                 >
                   <i className="ion-heart" />
-                  &nbsp; {faker.datatype.number()}
+                  &nbsp; {article?.favoritesCount}
                 </button>
               </div>
               <a className="preview-link">
-                <h1>{faker.lorem.sentence()}</h1>
-                <p>{faker.lorem.paragraph()}</p>
+                <h1>{article?.title}</h1>
+                <p>{article?.description}</p>
                 <span>Read more...</span>
                 <ul className="tag-list">
-                  <li className="tag-default tag-pill tag-outline">{faker.lorem.word()}</li>
+                  {article?.tagList?.map((tag) => { 
+                    return <li className="tag-default tag-pill tag-outline">{tag}</li>})}
                 </ul>
               </a>
             </div>
+            )})}
             <nav>
               <ul className="pagination">
-                <li className="page-item">
-                  <button type="button" className="page-link">
-                    1
-                  </button>
-                </li>
-                <li className="page-item">
-                  <button type="button" className="page-link">
-                    2
-                  </button>
-                </li>
-                <li className="page-item active">
-                  <button type="button" className="page-link">
-                    3
+                {Array.from({ length: articlesCount / DEFAULT_LIMIT}, (_, index) => { return ( 
+                <li className={classNames("page-item", {active: index === offset / DEFAULT_LIMIT})}>
+                  <button onClick={()=> setOffset(index * DEFAULT_LIMIT)} type="button" className="page-link">
+                    {index+1}
                   </button>
                 </li>
+                ) })}
+               
               </ul>
             </nav>
           </div>
           <div className="col-md-3">
-            <a href="#" className="tag-pill tag-default">
-              {faker.lorem.word()}
-            </a>
-            <a href="#" className="tag-pill tag-default">
-              {faker.lorem.word()}
-            </a>
-            <a href="#" className="tag-pill tag-default">
-              {faker.lorem.word()}
+            <div className="sidebar">
+              <p>Popular Tags</p>
+              <div className="tag-list">
+      
+            {tags?.map((tag) => { return (
+              <a onClick={()=> setActiveTag(tag)} href="#" className="tag-pill tag-default">
+              {tag}
             </a>
+            )})}
+            </div>
+            {tagsQuery.isFetching && <div>Loading Tags...</div>}
+            {(tags?.length === 0 && tagsQuery.isFetched) && <div>No tags are here... yet.</div>}
```

Comment:
Please use `isEmpty` from `lodash-es`.

Reply:
Fixed, please check.

1. For some reason prettier wont wort on auto save with VSCode just in this spesific project
2. I ran an 'npx prettier --write *' on the project folder for temp fix
3. For my first question - i understand the axios params as key/value but we only give it "offset" and not a real key/value base object. seems like it knows to transform it to one, is it for any variable or just for react states?

Reply:
Yeah, we went over this at the meeting tomorrow, there is a shorthand that was introduced with es6 where object properties that have a value with the same name can be collapsed.